ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/serde/ConnectSerdeSupplier.java

```diff
@@ -87,14 +88,30 @@ public byte[] serialize(final String topic, final Object spec) {
       }
 
       final T schema;
+      final int id;
       try {
         final String subject = KsqlConstants.getSRSubject(topic, isKey);
-        final int id = srClient.getLatestSchemaMetadata(subject).getId();
+        id = srClient.getLatestSchemaMetadata(subject).getId();
         schema = (T) srClient.getSchemaBySubjectAndId(subject, id);
       } catch (Exception e) {
         throw new KsqlException(e);
       }
 
+      // Do not let the serializer to auto register a new schema on JSON_SR formats.
+      // QTT tests with JSON_SR started failing on historic plans after supporting the
+      // JsonSchemaConverter because the converter was auto registering a schema with extra
+      // properties. These new props are not required, but was causing noise on the historic plans.
```

Comment:
Prod code when you insert still registers new schema for json_sr format, why we don't have problem there?

Reply:
It's caused by the historical plans with wrap singles = false. For instance, in 6.1, the SR schema used by the historical plan is `FOO STRUCT<ARRAY<STRING>>`; but when the wrap_single_value=false is used, then KSQL will attempt to serialize the schema `ARRAY<STRING>` which the SR API will try to register. SR will fail because the schema is not compatible with the latest schema that has a STRUCT.

All this is caused by QTT. I can explain you in more detail how QTT was working before offline. For QTT to work, I had to prevent QTT to register the schema as it was already registered before.